### DISCUSSION TITLE
How to clear cache for all the inputs of a Family provider at once?

### DISCUSSION THREAD

**Discussion by yogi7y** [rrousselGit/riverpod#1614] [Q&A] [answered]
How can we clear a family provider's cache for all of the different inputs at once? The StackOverflow [example](https://github.com/rrousselGit/riverpod/blob/master/examples/stackoverflow/lib/home.dart) can be used as a starting point:
```
RefreshIndicator(
    onRefresh: () {
        ref.refresh(paginatedQuestionsProvider(0));
        return ref.read(paginatedQuestionsProvider(0).future);
    }
)
```
This appears to clear the cache only for input 0; if I scroll down to page 1 or 2, for instance, I will see the cached result.

How to refresh only the first page and dispose of all the other inputs so that on scrolling down the API call is made and all the new results are returned? One naive approach that comes to my mind is to keep track of all the inputs and dispose of them iteratively when pull-to-refresh is called. I am looking for a better way to accomplish this. Thanks!

**Comment by rrousselGit** [comment, marked as answer]
There's an upcoming "ref.invalidate(family)"

It's available on master but not released yet

Le sam. 10 sept. 2022 à 10:27, Yogesh Parwani ***@***.***> a
écrit :

**Reply by yogi7y** [reply]
Awesome! Thanks, @rrousselGit

**Reply by jibbers42** [reply]
> There's an upcoming "ref.invalidate(family)" It's available on master but not released yet Le sam. 10 sept. 2022 à 10:27, Yogesh Parwani 

Is this expected to clear all data or am I using it wrong (it isn't clearing the data for me)?

```dart
@override
void dispose() {
  ref.invalidate(myFamilyProvider);
  super.dispose();
}
```
(riverpod 2.0.2)

EDIT:
Ugh, this question/answer format gets me every time. I'm going start a new one...

EDIT 2:
I mistook "upcoming" for v2, but just noticed he said it was on master and not yet released. I'll wait to try it after release before starting a new issue.
